Client/game_sa/CModelInfoSA.cpp

```diff
@@ -950,39 +950,50 @@ void CModelInfoSA::SetVehicleDummyPosition(eVehicleDummies eDummy, const CVector
         Request(BLOCKING, "SetVehicleDummyPosition");
 
     // Store default position in map
-    auto pVehicleModel = reinterpret_cast<CVehicleModelInfoSAInterface*>(m_pInterface);
-    auto iter = ms_ModelDefaultDummiesPosition.find(pVehicleModel);
+    auto iter = ms_ModelDefaultDummiesPosition.find(this);
     if (iter == ms_ModelDefaultDummiesPosition.end())
     {
-        ms_ModelDefaultDummiesPosition.insert({pVehicleModel, std::map<eVehicleDummies, CVector>()});
+        ms_ModelDefaultDummiesPosition.insert({this, std::map<eVehicleDummies, CVector>()});
         // Increment this model references count, so we don't unload it before we have a chance to reset the positions
         m_pInterface->usNumberOfRefs++;
     }
 
-    if (ms_ModelDefaultDummiesPosition[pVehicleModel].find(eDummy) == ms_ModelDefaultDummiesPosition[pVehicleModel].end())
+    auto pVehicleModel = reinterpret_cast<CVehicleModelInfoSAInterface*>(m_pInterface);
+    if (ms_ModelDefaultDummiesPosition[this].find(eDummy) == ms_ModelDefaultDummiesPosition[this].end())
     {
-        ms_ModelDefaultDummiesPosition[pVehicleModel][eDummy] = pVehicleModel->pVisualInfo->vecDummies[eDummy];
+        ms_ModelDefaultDummiesPosition[this][eDummy] = pVehicleModel->pVisualInfo->vecDummies[eDummy];
     }
 
     // Set dummy position
     pVehicleModel->pVisualInfo->vecDummies[eDummy] = vecPosition;
 }
 
-void CModelInfoSA::ResetAllVehicleDummies()
+void CModelInfoSA::ResetVehicleDummies()
 {
-    for (auto& info : ms_ModelDefaultDummiesPosition)
+    if (!IsVehicle())
+        return;
+
+    auto iter = ms_ModelDefaultDummiesPosition.find(this);
+    if (iter == ms_ModelDefaultDummiesPosition.end())
+        return; // Early out in case the model doesn't have any dummies modified
+
+    auto pVehicleModel = reinterpret_cast<CVehicleModelInfoSAInterface*>(m_pInterface);
+    for (auto& dummy : ms_ModelDefaultDummiesPosition[this])
     {
-        CVehicleModelInfoSAInterface* pVehicleModel = info.first;
-        for (auto& dummy : ms_ModelDefaultDummiesPosition[pVehicleModel])
-        {
-            // TODO: Find out why this is a nullptr, and fix underlying bug
-            if (pVehicleModel->pVisualInfo != nullptr)
-                pVehicleModel->pVisualInfo->vecDummies[dummy.first] = dummy.second;
-        }
-        ms_ModelDefaultDummiesPosition[pVehicleModel].clear();
-        // Decrement reference counter, since we reverted all position changes, the model can be safely unloaded
-        info.first->usNumberOfRefs--;
+        if (pVehicleModel->pVisualInfo != nullptr)

```

Comment:
I can confirm that this fix works without this line!

Reply:
Nice, should i remove it then? I can't be 100% sure that there are no other edge cases that turn this field into null.

Reply:
Perhaps we can log a line here instead. You can pick a number that isn't mentioned in this list: https://gist.github.com/qaisjp/fb9b2e78a04836c66344888a0161a003